test/solidity/Facets/DeBridgeDlnFacet.t.sol

```diff
@@ -0,0 +1,182 @@
+import { DeBridgeDlnFacet } from "lifi/Facets/DeBridgeDlnFacet.sol";
+import { IDlnSource } from "lifi/Interfaces/IDlnSource.sol";
+
+// Stub DeBridgeDlnFacet Contract
+contract TestDeBridgeDlnFacet is DeBridgeDlnFacet {
+    constructor(IDlnSource _dlnSource) DeBridgeDlnFacet(_dlnSource) {}
+
+    function addDex(address _dex) external {
+        LibAllowList.addAllowedContract(_dex);
+    }
+
+    function setFunctionApprovalBySignature(bytes4 _signature) external {
+        LibAllowList.addAllowedSelector(_signature);
+    }
+}
+
+contract DeBridgeDlnFacetTest is TestBaseFacet {
+    DeBridgeDlnFacet.DeBridgeDlnData internal validDeBridgeDlnData;
+    TestDeBridgeDlnFacet internal deBridgeDlnFacet;
+    IDlnSource internal DLN_SOURCE =
+        IDlnSource(0xeF4fB24aD0916217251F553c0596F8Edc630EB66);
+    uint256 internal FIXED_FEE;
+
+    function setUp() public {
+        customBlockNumberForForking = 19279222;
+        initTestBase();
+
+        deBridgeDlnFacet = new TestDeBridgeDlnFacet(DLN_SOURCE);
+        bytes4[] memory functionSelectors = new bytes4[](4);
+        functionSelectors[0] = deBridgeDlnFacet
+            .startBridgeTokensViaDeBridgeDln
+            .selector;
+        functionSelectors[1] = deBridgeDlnFacet
+            .swapAndStartBridgeTokensViaDeBridgeDln
+            .selector;
+        functionSelectors[2] = deBridgeDlnFacet.addDex.selector;
+        functionSelectors[3] = deBridgeDlnFacet
+            .setFunctionApprovalBySignature
+            .selector;
+
+        addFacet(diamond, address(deBridgeDlnFacet), functionSelectors);
+        deBridgeDlnFacet = TestDeBridgeDlnFacet(address(diamond));
+        deBridgeDlnFacet.addDex(ADDRESS_UNISWAP);
+        deBridgeDlnFacet.setFunctionApprovalBySignature(
+            uniswap.swapExactTokensForTokens.selector
+        );
+        deBridgeDlnFacet.setFunctionApprovalBySignature(
+            uniswap.swapTokensForExactETH.selector
+        );
+        deBridgeDlnFacet.setFunctionApprovalBySignature(
+            uniswap.swapETHForExactTokens.selector
+        );
+
+        setFacetAddressInTestBase(
+            address(deBridgeDlnFacet),
+            "DeBridgeDlnFacet"
+        );
+
+        // adjust bridgeData
+        bridgeData.bridge = "deBridgeDln";
+        bridgeData.destinationChainId = 137;
+
+        // produce valid DeBridgeDlnData
+        validDeBridgeDlnData = DeBridgeDlnFacet.DeBridgeDlnData({
+            receivingAssetId: 0x2791Bca1f2de4661ED88A30C99A7a9449Aa84174, // Polygon USDC
+            receiver: abi.encodePacked(USER_RECEIVER),
+            minAmountOut: (defaultUSDCAmount * 95) / 100
+        });
+
+        vm.label(address(DLN_SOURCE), "DLN_SOURCE");
+        FIXED_FEE = DLN_SOURCE.globalFixedNativeFee();
+    }
+
+    function initiateBridgeTxWithFacet(bool isNative) internal override {
+        if (isNative) {
+            validDeBridgeDlnData.minAmountOut =
```

Comment:
would be helpful to have a comment here explaining the rationale behind this code (why are we adding slippage here and why only for native?)